rotkehlchen/assets/utils.py

```diff
@@ -96,19 +120,21 @@ def get_or_create_evm_token(
             protocol=protocol,
             underlying_tokens=underlying_tokens,
         )
-        # This can but should not raise InputError since it should not already exist
-        # We need to wrap the whole call to add_ethereum_token_to_db since it makes
-        # the insert + when returning the EvmToken it can do a read to the database
-        globaldb = GlobalDBHandler()
-        with globaldb.conn.critical_section():
-            ethereum_token = add_ethereum_token_to_db(
-                token_data=token_data,
-                globaldb=globaldb,
-            )
-
-        # Also update the user db in a criticial section to avoid a switch
-        with userdb.conn.critical_section(), userdb.user_write() as cursor:
-            userdb.add_asset_identifiers(cursor, [ethereum_token.identifier])
+        if asset_exists is True:
+            # This means that we might need to update the information in the database
+            GlobalDBHandler().edit_evm_token(token_data)
```

Comment:
Should this be "might"?

```suggestion
            # This means that we need to update the information in the database with the newly queried data
            GlobalDBHandler().edit_evm_token(token_data)
```